pkg/resource/deploy/step_generator.go

```diff
@@ -38,6 +39,78 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/result"
 )
 
+// ContinueResourceDiffEvent is a step that asks the engine to continue provisioning a resource after completing its
+// diff, it is always created from a base RegisterResourceEvent.
```

Comment:
Nit but I feel like this is a bit weird coming straight in at the top of the file. Not sure if it has a more natural place?

Reply:
Moved to its own file